doc/dsdr/0010-placement-modelling.md

```diff
@@ -0,0 +1,105 @@
+
+![Segmentation](https://raw.githubusercontent.com/MastodonC/witan.cic/dsdrs/doc/img/segmentation.png)
+
+The constraint that all segments except terminal segments are sourced by segmenting existing periods into sequential 6-year chunks implies a potential issue completing an open case of duration 9 years: would we match to a segment captured starting after 6 years or after 12 years? Because neither may be a good match, we implement an 'offset' algorithm for splicing segments other than at a 6-year boundary. This algorithm starts by calculating the offset through the latest 6-year window of the end of the open case. For 9 years this might be 3 * 365 = 1095 days, because the 9th year has a 3 year offset into the second segment. We also create offset versions of all segments, skipping the first n days to create tail portions. In order to complete the matching efficiently, we pre-calculate all offset versions of all segments in advance. An offset version of a terminal segment is still a terminal segment.
+
+![Segmentation](https://raw.githubusercontent.com/MastodonC/witan.cic/dsdrs/doc/img/tail-segments.png)
+
+*Completing open cases*
+
+Completing an open case proceeds in a loop, matching a 6 year segment (or an offset equivalent), until we reach a terminal segment or exceed 18 years of age, at which point the loop terminates.
+
+![Segmentation](https://raw.githubusercontent.com/MastodonC/witan.cic/dsdrs/doc/img/offset-matching.png)
+
+Matching uses a combination of exact and fuzzy matching. Exact matching of certain fields ensures that segments being chained are compatible, whereas fuzzy matching ensures that the algorithm is non-deterministic and introduces useful variety into the generated ccandidates. Fields which are matched exactly are the placement at the join, the offset (measured in days) from a round 6 years and whether or not a 'terminal segment' is desired. Fuzzy matches are the join age in days and the number of days in care. Fuzzy matching is implemented by adding noise to the match values and then finding the closest matching segment using the Euclidean distance. Because match values of join age and days in care have well-defined min and max values (0 days and the durations corresponding to the 18th birthday respectively), binomial (instead of guassian) noise is added using `cic.model/jitter-binomial`. The binomial distribution of noise has several useful properties: it ensures that values are constrained within the min and max whilst remaining balanced around the mean, and has smaller variance at the extremes (close to zero and 18 years old) where timings are more sensitive and increasing variance in the middle of the range where timings are much less sensitive.
```

Comment:
Useful to have a pointer to the matching function

Reply:
I'm not sure which bit of the text you wanted the matching function for, but I've added references to `cic.model/markov-period` which does most of the work, and `cic.model/get-matched-segment` which is how the closest match is identified using the Euclidean distance.